examples/chip-tool/commands/payload/AdditionalDataParseCommand.cpp

```diff
@@ -26,22 +27,16 @@ using namespace ::chip::SetupPayloadData;
 
 CHIP_ERROR AdditionalDataParseCommand::Run()
 {
-    std::vector<uint8_t> payloadData;
     AdditionalDataPayload resultPayload;
     CHIP_ERROR err = CHIP_NO_ERROR;
-    std::string payloadString(mPayload);
 
-    // Decode input payload
-    size_t len = payloadString.length();
+    size_t additionalDataPayloadBytesLength = static_cast<size_t>(ceil(strlen(mPayload) / 2));
```

Comment:
`ceil` is a double method. This is not required here. The string is only allowed to be a multiple a two. Just check for multiple of 2 and bail out if not the case.